presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1046,7 +1046,6 @@ protected Type visitExists(ExistsPredicate node, StackableAstVisitorContext<Cont
         {
             StatementAnalyzer analyzer = statementAnalyzerFactory.apply(node);
             Scope subqueryScope = Scope.builder().withParent(scope).build();
-            analyzer.getAnalysis().setScope(node, subqueryScope);
```

Comment:
Scopes are required for nodes that produce relations. Expression doesn't produce relation, but rather value (or symbol).